### DISCUSSION TITLE
Success stories

### DISCUSSION THREAD

**Discussion by rrirower** [rrirower/DP2SceneGraphWizard#55] [General]
If you've been able to successfully convert your Direct Publisher channel to SceneGraph using this project, please indicate your experiences in this discussion.

**Comment by kf5ewy** [comment]
This project has helped me tremendously.  converting from direct publisher was  becoming a daunting task trying to learn how to build the channel from the ground up. It was not until I found your wizard that helped me so much. Not only to  successfully convert from Direct Publisher but also to see and learn more about the coding language that is used. Thank you @rrirower  for all your great work on this project and the  great help it has provided. keep up the good work looking forward to see where this project goes in the future.

**Comment by ferdiworks** [comment]
@rrirower I can honestly say that this project, This software has helped me with the automation of creating the template channel packages I needed to help convert multiple channels not only for myself but for some clients who needed simple conversions from Direct Publisher to SDK. I myself am a developer and can say that this has absolutely been a big time saver for me! I have been involved with testing this software from day 1 and each version update. I also added my own codes and enhancements to make it even work better throughout the process but also supported the developer and tried my best to assist and support some members using the software! Thank you for a great piece of software!

**Comment by adamesq** [comment]
A place just for success reporting! Sorry - I've been absent for a bit.

After several days stumbling around in murky code, I ran into DP2SceneGraphWizard & CFM. I got my channel switched over to SDK and approved.🙏🏻 THANK YOU SO MUCH to @rrirower (and @ferdiworks for adding in more than an extra 2 cents)...and this community!

Michael keeps it on the DL, but as someone who's benefited from the apps and seen the support & effort being put in, I'll urge others who may have missed the link...please BE SURE TO NOTE the donate button in the software if you're getting value from it. I think several of us would be up 💩's Creek without it...or at least losing many more stressful hours.

I've updated since, but JIC case it helps anyone...
SOFWARE VERSIONS I USED at the time:
DP2SGW version 1.0.0.0alpha 7.3 (or 7.2...can't quite remember - it's been a long few days on this)
CFM version 1.0.0.7
Note that I'm doing this on a Mac, using an instance of Windows via Amazon EC2 (first timer with that...just for these tools)
Contents in feed so far:
2 episode of 1 TV series
About 25 shortFormVideos (as of the original post, more at this point)

**Comment by HSBNDev** [comment]
This project really saved me and my organization from resorting to finding a 3rd party Developer to get this done. I am a one man show in my small non profit org and was handed the keys to our development while i am in college for Cybersecurity and have limited development experience outside of the classroom. 

Taking on the task of building the channel from scratch through Roku's very limited help with their own samples was proving to be a monumental task and i was getting very discouraged. Finding your project on the Roku Forums was a godsend and saved me and my org much time and money in getting this completed. Thank you so so much for your project.

The major hurdle i had using the tools was that my feed URL does not end in .json even though it is a json formatted feed which was developed by the previous person who built our tools before i was handed the keys to the kingdom. Because of this, the feed would get stuck at loading since it wasn't passing the URL .json check before launching into the feed parser function and I was able to correct the code for it to get into the parsing stage and everything worked beautifully once that was done. 

I got my first of 3 channels submitted and approved, deep linking code worked perfectly and was approved on the first try. Again, thank you so so much for this project and I already sent you a donation to keep you going on this as it really saved us.

**Comment by jsonnentag** [comment]
This wizard should be the first thing listed by Roku for people migrating from Direct Publisher to SceneGraph/SDK.

4 channels were just approved and I guess Deep Linking (as much as I listed) works.  There are no errors about it anyway.  BUT the thing I DID NOT know is that you have to add everything you want to Deep Link at the time of new channel version submission.  (???)  I was just going to add some more links and this message appeared:

Deep Linking
This page is not available until you upload a new package.

There are still some non-functional things in the wizard and files (Category "query" sorting and Overhangs not working in the latest rendition), but it is better than the "tutorial" from Roku that doesn't even tell you how to CREATE CATEGORIES.  I e-mailed Roku and last week there was a message back from them saying they had more messages than normal and that they would get back to me (ZERO HELP) and then another saying the exact same thing appeared today.  I asked them how to create Categories (since it isn't in the tutorial) and I guess they have a real hard time providing a useful answer on the topic. . . .

**Comment by eNEtv** [comment]
### Thank you @rrirower for an amazing project, your great work and support!

If anyone is curious how a channel based on this project may look, go to your **Roku** and search for the "**Explore New England**" channel. The ENE channel you will see, is based on this project v. 1.0.0.0-beta.3.1 where I included a few small code changes/updates.

Up to this day, Roku has been consistently failing to provide a complete working channel example. None of the Roku examples can pass certification validation so can't be published. Furthermore, their documentation is minimalist and hard to follow, with poor, incomplete examples. Having a background on mainstream programming languages, I find BrightScript not so bright, and Roku email support is far from useful. Sorry!

By contrast, this **gentleman** has put together a complete working project which can help you greatly, especially if you are not a dedicated/professional Roku programmer. 

Once again, many thanks!

**Comment by jasontalbott** [comment]
### This works so well!
I am still in development with my channel using this project but for the first time in a while I am enjoying working on our Roku channel. I had a Direct Publisher channel that did everything I needed and was very disappointed when ROKU shelved that platform. My currently published channel is using the template that ROKU made available to help people port their DP channels over to SceneGraph but I was always unsatisfied with having everything in one long row (17 tiles on the main screen to scroll through is not very user friendly.) So I was pleasantly surprised when in just a few minutes with @rrirower's wizard I was able to generate something 10 times better.  Completely agree with @jsonnentag that this should be the FIRST thing listed by Roku for people converting their Direct Publisher channels to SceneGraph. Just a little customization with the fonts and my channel will be ready to go. I'll come back here with an update when my package is approved.

His [Channel Feed Manager](https://github.com/rrirower/Channel-Feed-Manager) tool is great too for anyone needing help creating a feed file.

**Reply by jasontalbott** [reply]
Update: I'm really pleased with how my channel has turned out. I had to take some time away from the project to focus on more pressing projects but it was fun to come back to a month later, a project I truly enjoyed. After a little hiccup with Roku's "Channel behavior analysis" (their fault, nothing to do with this Wizard) my package passed all the tests and is awaiting publication in 5 days. The end result is so much more useful than the DP conversion template Roku provided. Thanks again @rrirower!

**Reply by jasontalbott** [reply]
Again, I only have positive things to say about this wizard tool. I have been so frustrated with the other aspects of dealing with Roku, that I feel that if it weren't for this tool, I would have just given up. Submitting a channel update and seeing it through to publication have been a challenge, Roku is kind of a nightmare platform to deal with in my experience. In addition to the issue I previously mentioned with their Channel Behavior Analysis tool being completely broken, I received email three times in the last 11 days from ROKU saying "Congratulations, your ROKU Channel passed all of our testing and has been published in the Channel Store..." when in reality some undisclosed bug in their process prevented publication and it took pleading with Partner Success to push it through. But no explanation from ROKU on what was going wrong on their end. On the other hand, the rare times I've found the Wizard to not quite work the way it should, @rrirower has been super transparent and super helpful and gotten things done! Makes me wish he worked for ROKU, they could really use his help, lol. 

Anyway, [channel published](https://channelstore.roku.com/details/6579a12ebe3c432a954b886b2bc210e1/tomorrows-world) for real today and people are starting to see it on the user end and the compliments are coming in.  We're forever grateful, thanks so much.

**Comment by JLNjolo** [comment]
As difficult to us "novice" tech people Roku can be to developers, I found that using the softwares created by @rrirower simplifies what would otherwise be a very difficult to understand task. Overall I've created 6 published Roku public channels and have heavily relied upon these amazing and well-thought through softwares and I am very grateful they are made available to the public. Thank you for your amazing work. You have made creating and converting channels much easier, and I hope that those who have benefited from your hard work, as a small "thank you" will take a moment to donate to your projects.
